src/components/admin/personen/PersonLock.vue

```diff
@@ -43,12 +86,27 @@
     return schulen.value.length <= 1;
   });
 
+  function resetBefristungFields(): void {
+    isUnbefristet.value = true;
+    formContext.resetField('selectedBefristung');
+    selectedBefristungProps.value.error = false;
+    selectedBefristungProps.value['error-messages'] = [];
+  }
+
   function closeLockPersonDialog(isActive: Ref<boolean>): void {
+    resetBefristungFields();
     isActive.value = false;
   }
-  function handleOnLockUser(id: string, isActive: Ref<boolean>): void {
+  async function handleOnLockUser(id: string, isActive: Ref<boolean>): Promise<void> {
+    if (!isUnbefristet.value) {
+      let { valid }: FormValidationResult<PersonLockForm, PersonLockForm> = await formContext.validate();
+      if (!valid) return;
+    }
+
     if (selectedSchule.value) {
-      emit('onLockUser', id, !props.person.person.isLocked, selectedSchule.value);
+      let dateISO: string | undefined = formatDateToISO(selectedBefristung.value);
```

Comment:
gehört zum Parent, bitte hier nur emitten, was eingegeben wird.  

Beispiel: **BefristungsInput**.

Reply:
Angepasst soweit es geht. Es gab aus main noch Änderungen in der View